src/main/kotlin/com/hibob/academy/employeeFeedback/dao/FeedbackDao.kt

```diff
@@ -38,4 +38,34 @@ class FeedbackDao(private val sql: DSLContext) {
             .execute() > 0
     }
 
+
+
+    fun getFeedbackByParams(companyId: Long, params: FeedbackQueryParams): List<FeedbackData> {
+        val query = sql.select(
+            feedbackTable.id,
+            feedbackTable.feedback,
+            feedbackTable.employeeId,
+            feedbackTable.companyId,
+            feedbackTable.date,
+            feedbackTable.status
+        )
+            .from(feedbackTable)
+            .leftJoin(employeeTable).on(feedbackTable.employeeId.eq(employeeTable.id))
+            .where(feedbackTable.companyId.eq(companyId))
+
+        params.department?.let {
+            query.and(employeeTable.department.eq(it.name))
+                .and(feedbackTable.employeeId.eq(employeeTable.id))
+        }
+
+        if (params.nullEmployeeId)
+            query.and(feedbackTable.employeeId.isNull)
+
+        params.date?.let {
+            query.and(feedbackTable.date.eq(params.date))
+        }
+
+        return query.fetch(feedbackMapper)
```

Comment:
query is immutable. calling the and function returns a new object and does not mutate the existing query AFAIK

Reply:
OTOH your tests are passing... weird